common/test_local_filestore.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for local_filestore.py."""
+
+import os
+import subprocess
+from unittest import mock
+
+import pytest
+
+from common import local_filestore
+
+
+def test_rm(tmp_path):
+    """Tests rm works as expected."""
+    file_path = tmp_path / 'file'
+    data = 'hello'
+    with open(file_path, 'w') as file_handle:
+        file_handle.write(data)
+    local_filestore.rm(str(file_path))
+    assert not os.path.exists(file_path)
+
+
+def test_ls(tmp_path):
+    """Tests ls will raise an exception while |must_exist| is
+    True if the file doesn't exist ."""
+    file_path = tmp_path / 'non_exist_file'
+    with pytest.raises(subprocess.CalledProcessError):
+        local_filestore.ls(str(file_path))
+
+
+def test_ls_non_must_exist(tmp_path):
+    """Tests ls won't raise an exception while |must_exist| is
+    False if the file doesn't exist ."""
+    file_path = tmp_path / 'non_exist_file'
+    local_filestore.ls(str(file_path), must_exist=False)
```

Comment:
Consider testing also if it returns what the docstring says it should return (ie. one filename at a line)

Reply:
Yes. this is a very good idea considering how much time was spent discussing how to get this righ.

Reply:
Done in #422. Thanks.